projects/angular-grid-layout/src/lib/grid.component.ts

```diff
@@ -350,21 +350,20 @@ export class KtdGridComponent implements OnChanges, AfterContentInit, AfterConte
         return new Observable<KtdGridLayout>((observer: Observer<KtdGridLayout>) => {
             // Retrieve grid (parent) and gridItem (draggedElem) client rects.
             const gridElemClientRect: KtdClientRect = getMutableClientRect(this.elementRef.nativeElement as HTMLElement);
-            const dragElemClientRect: KtdClientRect = getMutableClientRect(gridItem.elementRef.nativeElement as HTMLElement);
+            const dragElemClientRect: KtdClientRect = getMutableClientRect(gridItem.container.nativeElement as HTMLElement);
 
             const scrollableParent = typeof this.scrollableParent === 'string' ? document.getElementById(this.scrollableParent) : this.scrollableParent;
 
-            this.renderer.addClass(gridItem.elementRef.nativeElement, 'no-transitions');
-            this.renderer.addClass(gridItem.elementRef.nativeElement, 'ktd-grid-item-dragging');
-
-            // Create placeholder element. This element would represent the position where the dragged/resized element would be if the action ends
-            const placeholderElement: HTMLDivElement = this.renderer.createElement('div');
-            placeholderElement.style.width = `${dragElemClientRect.width}px`;
-            placeholderElement.style.height = `${dragElemClientRect.height}px`;
-            placeholderElement.style.transform = `translateX(${dragElemClientRect.left - gridElemClientRect.left}px) translateY(${dragElemClientRect.top - gridElemClientRect.top}px)`;
-
-            this.renderer.addClass(placeholderElement, 'ktd-grid-item-placeholder');
-            this.renderer.appendChild(this.elementRef.nativeElement, placeholderElement);
+            this.renderer.addClass(gridItem.container.nativeElement, 'no-transitions');
+            this.renderer.addClass(gridItem.container.nativeElement, 'ktd-grid-item-dragging');
+            
+            gridItem.placeholderContainer.nativeElement.style.width = `${dragElemClientRect.width}px`;
+            gridItem.placeholderContainer.nativeElement.style.height = `${dragElemClientRect.height}px`;
+            gridItem.placeholderContainer.nativeElement.style.transform = `translateX(${dragElemClientRect.left - gridElemClientRect.left}px) translateY(${dragElemClientRect.top - gridElemClientRect.top}px)`;
+            gridItem.placeholderContainer.nativeElement.style.display = 'block'
+            if (gridItem.placeholder && gridItem.placeholderContainer.nativeElement.children.length > 0) {
+                gridItem.placeholderContainer.nativeElement.style.opacity = 1
```

Comment:
This 'patch' would not be needed creating the placeholder on demand at grid level.